packages/frontend/src/pages/Template/components/GroupTemplateStepContent.tsx

```diff
@@ -54,19 +57,19 @@ export default function IfThenTemplateStepContent(
       borderRadius="lg"
       pb={4}
     >
-      {/* This is for the if then header step */}
+      {/* This is for the group header step */}
       <Flex w="100%" p={4} alignItems="center" columnGap={4}>
         <Icon boxSize={6} as={BiGitRepoForked} color="primary.500" ml={2} />
 
-        <Text textStyle="subhead-1">If-then</Text>
+        <Text textStyle="subhead-1">{header}</Text>
```

Comment:
think can add a variable to check for which icon to display for if-then and for-each, similar to the header message